reframe/core/pipeline.py

```diff
@@ -138,33 +138,49 @@ class RegressionTest(RegressionMixin, jsonext.JSONSerializable):
     This class provides the implementation of the pipeline phases that the
     regression test goes through during its lifetime.
 
-    :param require_version: is a list of ReFrame version specifications that
-        this test is allowed to run.
-        A version specification string can have one of the following formats:
-
-        1. ``VERSION``: Specifies a single version.
-        2. ``{OP}VERSION``, where ``{OP}`` can be any of ``>``, ``>=``, ``<``,
-            ``<=``, ``==`` and ``!=``. For example, the version specification
-            string ``'>=3.5.0'`` will allow the following test to be loaded
-            only by ReFrame 3.5.0 and higher. The ``==VERSION`` specification
-            is the equivalent of ``VERSION``.
-        3. ``V1..V2``: Specifies a range of versions.
+    This class accepts parameters at the *class definition*, i.e., the test
+    class can be defined as follows:
+
+    .. code-block:: python
+
+       class MyTest(RegressionTest, param='foo', ...):
+
+    where ``param`` is one of the following:
+
+    :param pin_prefix: the prefix to pin the test to if the test lives in a
+        test library with resources in it.
```

Comment:
To me this reads as if this parameter requires an actual prefix instead of being a bool flag. How about something like "Lock the test prefix to the directory where the current class lives."?